maptivist/src/main/webapp/script.js

```diff
@@ -176,17 +176,39 @@ function createInfowindow(position) {
         <p>counter: </p>
     </div>
     <div class="flag">
-        <button>Flag</button>
+        <button onclick = "flagFunction()">Flag</button>
```

Comment:
So, I would pass the marker into `createInfoWindow(position, marker)`

Then for the button, add an `id="info_window_flag_button"`:

And then you can add a `domready` listener to the infowindow:
https://developers.google.com/maps/documentation/javascript/reference/3.40/info-window?hl=en#InfoWindow.domready

Finally, add a `domready` listener
```
infowindow.domready = function() {
   var flag_btn =   document.getElementById("info_window_flag_button");
   flag_btn.onclick = function() {
      // the closure captures the marker.
      flagFunction(marker);
   }
}
```

I *think* something like that should work.

Reply:
I think it does work, but until markers can be successfully created I won't know for sure. I've committed and pushed the changes though and will return to this PR when markers can be successfully implemented